plugins/bgp/bgp-plugin.go

```diff
@@ -40,12 +63,128 @@ func (p *BgpPlugin) Init() error {
 	p.KVScheduler.RegisterKVDescriptor(gd)
 
 	// register descriptor for bgp peer config
+
 	pd := descriptor.NewPeerConfDescriptor(p.Log, p.BGPServer)
 	p.KVScheduler.RegisterKVDescriptor(pd)
 
+	p.watchCloser = make(chan string)
+	watcher := p.Deps.KVStore.NewWatcher(nodePrefix)
+	err := watcher.Watch(p.onChange, p.watchCloser, "")
+	if err != nil {
+		return err
+
+	}
 	return nil
 }
 func (p *BgpPlugin) Close() error {
 	log.Println("Closing BgpPlugin Application.")
 	return nil
 }
+
+//tutorial says keyval.protowatchresp but i couldnt find it
+//in the  cn-infra/db/keyval/proto_watcher_api.go in our vendor folder there isnt a ProtoWatchResp
+// but on the github for cn infra, there is one
+func (p *BgpPlugin) onChange(resp datasync.ProtoWatchResp) {
+	//key := resp.GetKey()
+
+	//Getting
+	value := new(vppnode.VppNode)
+	changeType := resp.GetChangeType()
+	if changeType == datasync.Delete {
+		if prevValExist, err := resp.GetPrevValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+		} else if !prevValExist {
+			logging.Errorf("Delete and Previous Value Does not Exist")
+		}
+	} else {
+		if err := resp.GetValue(value); err != nil {
+			logging.Errorf("get value error: %v", err)
+		}
+	}
+	ip := value.IpAddresses[0]
+	ipParts := strings.Split(ip, "/")
+	ip = ipParts[0]
+	//prefixlen, err4 := strconv.Atoi(ipParts[1])
+
+	client, err := remote.CreateHTTPClient("")
+	if err != nil {
+		logging.Error("create http client error: %v", err)
+	}
+
+	//Getting Ipam Info
+	b, err2 := getNodeInfo(client, ip, getIpamDataCmd)
+	if err2 != nil {
+		logging.Errorf("getnodeinfo error: %v", err2)
+	}
+
+	ipam := ipv4net.IPAMData{}
+	err3 := json.Unmarshal(b, &ipam)
+	if err3 != nil {
+		logging.Errorf("unmarshal error: %v", err3)
+	}
+
+	//Setting Route info
+	nlri, _ := ptypes.MarshalAny(&bgp_api.IPAddressPrefix{
+		Prefix:    ipam.PodSubnetThisNode,
+		PrefixLen: 24,
+	})
+
+	a1, _ := ptypes.MarshalAny(&bgp_api.OriginAttribute{
+		Origin: 0,
+	})
+	a2, _ := ptypes.MarshalAny(&bgp_api.NextHopAttribute{
+		NextHop: ip,
+	})
+	attrs := []*any.Any{a1, a2}
+	if changeType == datasync.Put {
```

Comment:
use a switch statement instead of multiple if/else - it's kind of easier to read